tests/test_main.py

```diff
@@ -0,0 +1,265 @@
+
+    @patch('homu.main.PullReqState')
+    def test_retry_approved(self, MockPullReqState):
+        state = MockPullReqState()
+        state.try_ = False
+        retry(state, True)
+        state.set_status.assert_called_once_with('')
+        state.change_labels.assert_called_once_with(LabelEvent.APPROVED)
+
+    @patch('homu.main.PullReqState')
+    def test_delegate_negative(self, MockPullReqState):
+        state = MockPullReqState()
+        delegate_negative(state)
+        self.assertEqual(state.delegate, '')
+        state.save.assert_called_once()
+
+    @patch('homu.main.PullReqState')
+    def test_delegate_positive(self, MockPullReqState):
+        state = MockPullReqState()
+        delegate_positive(state, True)
+        state.get_repo.assert_called_once()
+        state.add_comment.assert_called_once()
+        state.save.assert_called_once()
+
+    @patch('homu.main.PullReqState')
+    def test_delegate_to(self, MockPullReqState):
+        state = MockPullReqState()
+        delegate_to(state, True, 'delegate=user')
+        self.assertEqual(state.delegate, 'user')
+        state.save.assert_called_once()
+        state.add_comment.assert_called_once_with(
+            ':v: @user can now approve this pull request'
+        )
+
+    @patch('homu.main.PullReqState')
+    def test_set_priority_not_priority_less_than_max_priority(self, MockPullReqState):
+        state = MockPullReqState()
+        set_priority(state, True, 'p=1', {'max_priority': 3})
+        self.assertEqual(state.priority, 1)
+        state.save.assert_called_once()
+
+    @patch('homu.main.PullReqState')
+    def test_set_priority_not_priority_more_than_max_priority(self, MockPullReqState):
+        state = MockPullReqState()
+        set_priority(state, True, 'p=5', {'max_priority': 3})
+        state.add_comment.assert_called_once_with(':stop_sign: Priority higher than 3 is ignored.')
+
+    @patch('homu.main.PullReqState')
+    def test_review_approved_approver_me(self, MockPullReqState):
+        state = MockPullReqState()
+        self.assertFalse(review_approved(['r=me', 'approved'], 'r=me', state, True, 'user', 'user', 0, '', []))
+
+    @patch('homu.main.PullReqState')
+    def test_review_approved_wip(self, MockPullReqState):
+        state = MockPullReqState()
+        state.title = 'WIP work in progress'
+        self.assertFalse(review_approved(['r+', 'approved'], 'r+', state, True, 'user', 'user', 0, '', []))
+        state.add_comment.assert_called_once_with(':clipboard: Looks like this PR is still in progress, ignoring approval')
+
+    @patch('homu.main.PullReqState')
+    def test_review_approved_todo(self, MockPullReqState):
+        state = MockPullReqState()
+        state.title = 'TODO work in progress'
+        self.assertFalse(review_approved(['r+', 'approved'], 'r+', state, True, 'user', 'user', 0, '', []))
+        state.add_comment.assert_called_once_with(':clipboard: Looks like this PR is still in progress, ignoring approval')
```

Comment:
This seems like it could be merged with the previous test to reduce duplication.